mozilla-services/socorro
# bug 1603308: fix verifyprocessed

This fixes a couple of things in verifyprocessed command:

1. It fixes the day it checks. Before it would check "today", but since it ran at 4:00am, there wasn't much to check, so it probably hasn't been checking everything in ages. Plus since it's checking "today", it's picking up some raw crashes that are being processed and thinks they're missing. Bleh.

2. It adds checking for Elasticsearch and considers a crash missing if it's either not on S3 or not in Elasticsearch. This will plug a hole where some things could end up on S3, but due to bugs in the Elasticsearch crashstorage code, never make it to Elasticsearch and thus never be searchable. Bleh.

In the process of doing that, I had to rework how the tests work with Elasticsearch. The code changes are much better. While doing that, I hit bugs in how the webapp tests mock the socorro middleware model class implementations, so I fixed that, too.

To test:

1. run `make setup` to clear out all storage things
2. run `./scripts/process_crashes.sh CRASHID` which will pull down crash data and put the raw crash in S3--don't run the processor, yet
3. run `cd webapp; ./manage.py verifyprocessed` which will see that as a missing crash
4. run the processor
5. run `socorro-cmd es delete` to and then `socorro-cmd es create` to wipe out Elasticsearch storage so now the crash is on S3 but not Elasticsearch
6. run `cd webapp; ./manage.py verifyprocessed` which will see that the crash is still missing
7. process the crash
8. run `cd webapp; ./manage.py verifyprocessed` which will see that the crash is no longer missing--yay!
